src/main/java/seedu/address/model/consultation/ConsultationList.java

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * A list of Consultation.
+ */
+public class ConsultationList implements Iterable<Consultation> {
+    private ObservableList<Consultation> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Consultation> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    public void setConsultationList(List<Consultation> consultationList) {
+        requireAllNonNull(consultationList);
+        internalList.setAll(consultationList);
+    }
+
+    /**
+     * Returns true if the list contains an equivalent consultation as the given argument.
+     */
+    public boolean contains(Consultation toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameConsultation);
+    }
+
+    /**
+     * Adds a consultation to the list.
+     */
+    public void addConsultation(Consultation toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateConsultationException();
+        }
+        internalList.add(toAdd);
+    }
+
+    //    /**
+    //     * Replaces the consultation {@code target} in the list with {@code editedConsultation}.
+    //     * {@code target} must exist in the list.
+    //     * The consultation {@code editedConsultation} must not be the same as another existing
+    //     * consultation in the list.
+    //     */
+    //    public void setConsult(Consultation target, Consultation editedConsultation) {
+    //        requireAllNonNull(target, editedConsultation);
+    //
+    //        int index = internalList.indexOf(target);
+    //        if (index == -1) {
+    //            throw new ConsultationNotFoundException();
+    //        }
+    //
+    //        if (!target.isSameConsultation(editedConsultation) && contains(editedConsultation)) {
+    //            throw new DuplicateConsultationException();
+    //        }
+    //
+    //        internalList.set(index, editedConsultation);
+    //    }
+
+    /**
+     * Removes the equivalent consultation from the list.
+     * The consultation must exist in the list.
+     */
+    public void remove(Consultation toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new ConsultationNotFoundException();
+        }
+    }
+
+    public ObservableList<Consultation> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<Consultation> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public String toString() {
+        return internalList.toString();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true; // Same object, equal
+        }
+
+        if (!(other instanceof ConsultationList)) {
+            return false; // Not the same class, not equal
```

Comment:
same as for consultationlistbook, you can test this line by using ConsultationList.equals(*some other stuff*)

Reply:
Resolved thanks!